apps/rush-lib/src/cli/actions/BaseRushAction.ts

```diff
@@ -11,19 +12,43 @@ import {
 import RushConfiguration from '../../data/RushConfiguration';
 import EventHooksManager from '../logic/EventHooksManager';
 
+export interface IRushCommandLineActionOptions extends ICommandLineActionOptions {
+  /** If true, no locking mechanism will be enforced when this action is run */
+  safeForSimultaneousRushProcesses?: boolean;
+}
+
 /**
  * The base Rush action that all Rush actions should extend.
  */
 export abstract class BaseRushAction extends CommandLineAction {
   private _rushConfiguration: RushConfiguration;
   private _eventHooksManager: EventHooksManager;
+  private _safeForSimultaneousRushProcesses: boolean;
 
-  constructor(options: ICommandLineActionOptions) {
+  constructor(options: IRushCommandLineActionOptions) {
     super(options);
+    this._safeForSimultaneousRushProcesses = !!options.safeForSimultaneousRushProcesses;
   }
 
   protected onExecute(): void {
     this._ensureEnvironment();
+
+    if (!this._safeForSimultaneousRushProcesses) {
+      const lockFilePath: string = path.join(
+        this.rushConfiguration.commonTempFolder,
+        'rush.lock'
+      );
+      if (fsx.existsSync(lockFilePath)) {
```

Comment:
>existsSync [](start = 14, length = 10)

This file locking implementation is probably going to expand over time and include workarounds for specific operating systems.  It's a general pattern we'll want to use in other tools.

Could you move it to `node-core-library` and write some unit tests?